docs/setup/upgrade/resolving-migration-failures.asciidoc

```diff
@@ -123,3 +123,64 @@ In {kib} 7.5.0 and earlier, when the task manager index is set to `.tasks`
 with the configuration setting `xpack.tasks.index: ".tasks"`,
 upgrade migrations fail. In {kib} 7.5.1 and later, the incompatible configuration
 setting prevents upgrade migrations from starting.
+
+[float]
+==== Repeated time-out requests that eventually fail
+
+Migrations get stuck in a loop of retry attempts waiting for index yellow status that's never reached. If you see a log entry similar to `Action failed with "Request timed out". Retrying attempt 1 in 2 seconds.` in either the CLONE_TEMP_TO_TARGET or CREATE_REINDEX_TEMP steps, it's because the process is waiting for an index yellow status.
+
+There are two known potential causes:
+
+* cluster hits the low watermark for disk usage
+* cluster has <<routing-allocation-disabled,routing allocation disabled>>
+
+Before retrying the migration, inspect the output of the `_cluster/allocation/explain?index=${targetIndex}` API to identify what's preventing the index going yellow:
+
+[source,sh]
+--------------------------------------------
+GET _cluster/allocation/explain
+{
+  "index": ".kibana_8.1.0_001",
+  "shard": 0,
+  "primary": true,
+}
+--------------------------------------------
+
+If the cluster exceeded the low watermark for disc usage, the output should contain a message like 
+[source,sh]
+--------------------------------------------
+"the node is above the low watermark cluster setting [cluster.routing.allocation.disk.watermark.low=85%], using more disk space than the maximum allowed [85.0%], actual free: [11.692661332965082%]"
+--------------------------------------------
+
+Refer to the {es} guide on how to {ref}/fix-common-cluster-issues.html#_error_disk_usage_exceeded_flood_stage_watermark_index_has_read_only_allow_delete_block[fix common cluster issues].
+
+If routing allocation is the issue, you will should see an entry like
+[source,sh]
+--------------------------------------------
+"allocate_explanation" : "cannot allocate because allocation is not permitted to any of the nodes"
+--------------------------------------------
+
+[float]
+[[routing-allocation-disabled]]
+==== Routing allocation disabled or restricted
+Upgrade migrations fail because routing allocation is disabled or restricted (cluster.routing.allocation.enable: none/primaries/new_primaries), which causes {kib} to log errors such as:
```

Comment:
```suggestion
Upgrade migrations fail because routing allocation is disabled or restricted (`cluster.routing.allocation.enable: none/primaries/new_primaries`), which causes {kib} to log errors such as:
```